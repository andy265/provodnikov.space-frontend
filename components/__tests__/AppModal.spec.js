const { createRenderer } = require('vue-server-renderer')
const { shallow } = require('vue-test-utils')
const AppModal = require('~/components/AppModal')

describe('AppModal.vue', () => {
  let wrapper = shallow(AppModal, {
    propsData: {
      name: 'modal'
    },
    slots: {
      content: '<div>some content</div>',
      buttons: '<div>some buttons</div>'
    }
  })

  it('has same HTML structure', () => {
    let renderer = createRenderer()
    // eslint-disable-next-line
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })

  it('emit before-open event', () => {
    wrapper.vm.onBeforeOpen({
      params: {}
    })
    expect(wrapper.emitted()['before-open']).toBeTruthy()
  })

  it('emit opened event', () => {
    wrapper.vm.onOpened()
    expect(wrapper.emitted().opened).toBeTruthy()
  })

  it('emit closed event', () => {
    wrapper.vm.onClosed()
    expect(wrapper.emitted().closed).toBeTruthy()
  })
})
