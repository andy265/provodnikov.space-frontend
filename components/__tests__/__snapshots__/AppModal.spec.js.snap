// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppModal.vue has same HTML structure 1`] = `
<modal name="modal" adaptive="true" scrollable="true" reset="true" width="640px" height="auto" class="app-modal">
    <div class="app-modal__content">
        <div>some content</div>
    </div>
    <div class="app-modal__buttons">
        <div>some buttons</div>
    </div>
</modal>
`;
